test/src/commonMain/kotlin/io/realm/util/Utils.kt

```diff
@@ -17,10 +17,16 @@
 
 package io.realm.util
 
+import kotlin.time.Duration
+import kotlin.time.ExperimentalTime
+
 // Platform dependant helper methods
 expect object PlatformUtils {
     fun createTempDir(): String
     fun deleteTempDir(path: String)
+    @OptIn(ExperimentalTime::class)
+    fun sleep(duration: Duration)
```

Comment:
why not use coroutine delay?

Reply:
Delay is suspending and we need some primitive to allow write blocks to continue while we are progressing the tests from outside (cancelling, scheduling other writes, etc.)